### PR TITLE
resource/aws_amplify_branch: fix forces replacement

### LINKED ISSUE #34130: [Bug]: changing enable_performance_mode on aws_amplify_branch forces replacement
### Terraform Core Version

1.5.5

### AWS Provider Version

5.20.0

### Affected Resource(s)

* aws_amplify_branch

### Expected Behavior

Changing `enable_performance_mode` on an `aws_amplify_branch` should cause an update to the resource.

It should be possible because AWS CLI allows it:

```
aws amplify update-branch --app-id APP_ID --branch-name BRANCH_NAME --enable-performance-mode
```

### Actual Behavior

Changing `enable_performance_mode` on an `aws_amplify_branch` forces replacement

### Relevant Error/Panic Output Snippet

```shell
# module.frontend.aws_amplify_branch.branches["REDACTED"] must be replaced
-/+ resource "aws_amplify_branch" "branches" {
      ~ arn                         = "REDACTED" -> (known after apply)
      ~ associated_resources        = [] -> (known after apply)
      ~ custom_domains              = [] -> (known after apply)
      + destination_branch          = (known after apply)
      ~ display_name                = "REDACTED" -> (known after apply)
      - enable_basic_auth           = false -> null
      - enable_notification         = false -> null
      ~ enable_performance_mode     = false -> true # forces replacement
      - environment_variables       = {} -> null
      ~ id                          = "REDACTED" -> (known after apply)
      + source_branch               = (known after apply)
      - tags                        = {} -> null
      ~ tags_all                    = {} -> (known after apply)
      - ttl                         = "5" -> null
        # (6 unchanged attributes hidden)
    }
```

### Terraform Configuration Files

N/A

### Steps to Reproduce

* Add `resource "aws_amplify_app" "app"`
* Add `resource "aws_amplify_branch" "branch"` for the app with `enable_performance_mode = false`
* Deploy with `terraform apply`
* Change `enable_performance_mode = true` in `aws_amplify_branch.branch`
* Run `terraform plan` to see that it forces replacement.

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by naruse666**
### Description
Unnecessary `ForceNew` are included at `enable_performance_mode`.

### Relations
Closes #34130

### References
`ForceNew` isn't written in document.
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/amplify_branch#enable_performance_mode

### Output from Acceptance Testing
```console
$ make testacc TESTARGS='-run=TestAccAmplify_serial/Branch/OptionalArguments' PKG=amplify
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/amplify/... -v -count 1 -parallel 20  -run=TestAccAmplify_serial/Branch/OptionalArguments -timeout 360m
=== RUN   TestAccAmplify_serial
=== PAUSE TestAccAmplify_serial
=== CONT  TestAccAmplify_serial
=== RUN   TestAccAmplify_serial/Branch
=== RUN   TestAccAmplify_serial/Branch/OptionalArguments
--- PASS: TestAccAmplify_serial (338.48s)
    --- PASS: TestAccAmplify_serial/Branch (338.48s)
        --- PASS: TestAccAmplify_serial/Branch/OptionalArguments (338.47s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/amplify	343.023s
```

### CODE REVIEW

File: internal/service/amplify/branch.go [hunk 1]
```diff
@@ -126,7 +126,6 @@ func ResourceBranch() *schema.Resource {
 			"enable_performance_mode": {
 				Type:     schema.TypeBool,
 				Optional: true,
-				ForceNew: true,
```
Comment (KholdStare) [inline internal/service/amplify/branch.go:129]: Are the underlying API calls to AWS verified to work with this change?

Comment (naruse666) [reply internal/service/amplify/branch.go:129]: yes, it seems to be ok.
```console
$ terraform plan
  # aws_amplify_branch.master will be updated in-place
  ~ resource "aws_amplify_branch" "master" {
      ~ enable_performance_mode     = false -> true
        id                          = "d2rpmmwfwa1zzl/master"
        tags                        = {}
        # (14 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

**Comment by naruse666** [comment]
I found another bug.
had specified `EnablePullRequestPreview` on enable_performance_mode Update.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccAmplify_serial/Branch' PKG=amplify
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/amplify/... -v -count 1 -parallel 20  -run=TestAccAmplify_serial/Branch -timeout 360m
=== RUN   TestAccAmplify_serial
=== PAUSE TestAccAmplify_serial
=== CONT  TestAccAmplify_serial
=== RUN   TestAccAmplify_serial/Branch
=== RUN   TestAccAmplify_serial/Branch/disappears
=== RUN   TestAccAmplify_serial/Branch/tags
=== RUN   TestAccAmplify_serial/Branch/BasicAuthCredentials
    branch_test.go:150: Step 1/4 error: Check failed: Check 2/3 error: aws_amplify_branch.test: Attribute 'basic_auth_credentials' expected "dXNlcm5hbWUxOnBhc3N3b3JkMQ==", got "dXNlcm5hbWUxOk5nUmI0VW5IVytqQjkwZ2I4djJKUEE9PXx8MXRlRXByY1NBTmhMN1VQbGZ0QjZxMmduVmd5U0lXT3pldm1xR1hJZTBFND0="
=== RUN   TestAccAmplify_serial/Branch/EnvironmentVariables
=== RUN   TestAccAmplify_serial/Branch/OptionalArguments
=== RUN   TestAccAmplify_serial/Branch/basic
--- FAIL: TestAccAmplify_serial (252.41s)
    --- FAIL: TestAccAmplify_serial/Branch (252.41s)
        --- PASS: TestAccAmplify_serial/Branch/disappears (27.82s)
        --- PASS: TestAccAmplify_serial/Branch/tags (70.99s)
        --- FAIL: TestAccAmplify_serial/Branch/BasicAuthCredentials (13.54s)
        --- PASS: TestAccAmplify_serial/Branch/EnvironmentVariables (64.57s)
        --- PASS: TestAccAmplify_serial/Branch/OptionalArguments (46.95s)
        --- PASS: TestAccAmplify_serial/Branch/basic (28.54s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/amplify	258.431s
FAIL
make: *** [testacc] Error 1
```

Failure is unrelated to this change.

**Comment by ewbankkit** [comment]
@naruse666 Thanks for the contribution :tada: :clap:.

**Pull request merged**